fabric-input-events-v1/src/main/java/net/fabricmc/fabric/impl/client/input/InputCallbacksImpl.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.client.input;
+
+import java.util.Map;
+
+import org.lwjgl.glfw.GLFW;
+import org.lwjgl.glfw.GLFWDropCallback;
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+
+import net.minecraft.client.options.KeyBinding;
+import net.minecraft.client.util.InputUtil;
+import net.minecraft.client.util.InputUtil.Key;
+
+import net.fabricmc.fabric.api.event.client.input.ClientInputEvents;
+import net.fabricmc.fabric.mixin.event.input.client.accessor.InputUtilTypeAccessor;
+import net.fabricmc.fabric.mixin.event.input.client.accessor.KeyBindingAccessor;
+
+public final class InputCallbacksImpl {
+	public static void onKey(long window, int code, int scancode, int action, int modKeys) {
+		FabricKeyboardImpl.updateModKeys(modKeys);
+
+		ClientInputEvents.KEY.invoker().onKeyChanged(code, scancode, action, modKeys);
+
+		boolean hasListeners = ClientInputEvents.KEY_PRESSED.hasListeners()
+							|| ClientInputEvents.KEY_RELEASED.hasListeners()
+							|| ClientInputEvents.KEY_REPEATED.hasListeners()
+							|| ClientInputEvents.KEYBIND_PRESSED.hasListeners()
+							|| ClientInputEvents.KEYBIND_RELEASED.hasListeners()
+							|| ClientInputEvents.KEYBIND_REPEATED.hasListeners();
```

Comment:
I would usually use spaces here so it would look the same even without `.editorconfig`. I'm assuming this is fine considering `.editorconfig` has `tab_width = 4`.

Reply:
The general style used through Fabric API's codebase is to always use two tabs for continuation indent in places like this, not five. (Using tabs for visual alignment is unreliable as seen here.)

Reply:
> Using tabs for visual alignment is unreliable as seen here

Yep, I configured my editors to always use spaces for visual alignment for exactly this reason, but the style checker didn't allow it. I just regex replaced them and didn't notice this until now.